src/helpers/assistant/assistant.ts

```diff
@@ -0,0 +1,169 @@
+        this.additionalInstructions = additionalInstructions;
+        this.tools = tools.length ? (appendTools ? this.defaultTools.concat(tools) : tools) : this.defaultTools;
+    }
+
+    async ask(userMessage: string, onEvent: (event: any) => void) {
+
+        if (!this.initialized) {
+            let participantInfo = await MyDataHelps.getParticipantInfo();
+            this.initialize(participantInfo);
+        }
+
+        let config = { configurable: { thread_id: this.participantId } };
+        let inputs = {
+            messages: [
+                new HumanMessage(userMessage)
+            ]
+        };
+        for await (
+            const event of await this.graph.streamEvents(inputs, {
+                ...config,
+                streamMode: "values",
+                version: "v1"
+            })
+        ) {
+            onEvent(event);
+        }
+    }
+
+    private initialize(participantInfo: ParticipantInfo) {
+
+        const toolNode = new ToolNode<{ messages: BaseMessage[] }>(this.tools);
+
+        const graphState: StateGraphArgs<IAssistantState>["channels"] = {
+            messages: {
+                reducer: messagesStateReducer
+            },
+            participantInfo: {
+                value: (x: string, y: string) => y ? y : x,
+                default: () => "{}"
+            }
+        };
+
+        const boundModel = new ChatOpenAI({
+            model: "gpt-4o",
+            temperature: 0,
+            apiKey: MyDataHelps.token.access_token
+        }, {
+            baseURL: this.baseUrl
+        }).bindTools(this.tools);
+
+        const promptTemplate = ChatPromptTemplate.fromMessages([
+            SystemMessagePromptTemplate.fromTemplate(`
+			You are a health and wellness data assistant. Your purpose is to help users understand their health and wearable data,
+            which includes providing simple summaries and highlighting insights and connections between data. The tone should
+            be clear and friendly. You are not a coach, so should not give advice. You should not be disparaging or discouraging,
+            just objectively share information.
+            
+            You can encourage the user to ask more questions and even suggest additional follow-up questions that might be relevant.
+
+            If the user asks for some data, and you query it with a particular tool, but the returned data does not sufficiently
+            answer the user's question, for example, if the user asks for their last 3 LDL values, and you query the getEhrNewsFeedPage
+            tool and it returns only the last 2 LDL values and a nextPageID, then query the same tool again while passing the nextPageID as the
+            pageID parameter to fetch additional data. Continue this process until you have all the data that the user has asked for, up to 5 iterations.
+
+			User information: {participantInfo}
```

Comment:
I would also consider embedding the projectInfo here (the stuff you get via MyDataHelps.getProjectInfo() or whatever the method is) - as that could be quite useful for things like the support contact number, etc.

Reply:
(And is fast and takes no parameters, so doesn't make sense to be a tool)

Reply:
Good idea. Added.

In some future iteration I wonder if we should query upfront `getAllDailyDataTypes` and `getDeviceDataV2AllDataTypes` as well.

Reply:
Likely so, those are very fast calls so there is no downside to doing so